weather_test.go

```diff
@@ -0,0 +1,48 @@
+package weather_test
+
+import (
+	"os"
+	"testing"
+	"weather"
+)
+
+func TestConstructUrl(t *testing.T) {
+
+	token := "foo"
+	got, err := weather.Request([]string{"PATH", "rio", "de", "janeiro"}, token)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	want := "https://api.openweathermap.org/data/2.5/weather?q=rio%20de%20janeiro&appid=foo"
+
+	if want != got {
+		t.Fatalf("want: %q got: %q", want, got)
+	}
+
+}
+
+func TestParseResponseWeather(t *testing.T) {
+
+	file, err := os.Open("testdata/weather.json")
```

Comment:
If you change `ParseResponse` to take a `[]byte`, then you can change this test to use `os.ReadFile` instead (which returns a `[]byte, error`).